docs/docs/adapters/xata.md

```diff
@@ -0,0 +1,216 @@
+---
+id: xata
+title: Xata
+---
+
+# Xata
+
+This adapter is intended to be used with Xata projects. The preferred way to create a Xata project and use Xata databases is using the [Xata CLI](https://docs.xata.io/cli/getting-started). The CLI allows generating a `XataClient` that will help you work with Xata in a safe way, and that this adapter depends on.
+
+<!-- @todo add GIFs -->
+
+```bash npm2yarn2pnpm
+npm install next-auth @next-auth/xata-adapter
+npm install --location=global @xata.io/cli
+xata auth login
+xata init
+```
+
+Configure your `./pages/api/auth/[...nextauth]` route to use the Xata adapter:
+
+```typescript title="pages/api/auth/[...nextauth].ts"
+import NextAuth from "next-auth"
+import GoogleProvider from "next-auth/providers/google"
+import { XataAdapter } from "@next-auth/xata-adapter"
+import { XataClient } from "../../../../xata.ts" // or wherever you've chosen to create the client
+
+const client = new XataClient()
+
+export default NextAuth({
+  adapter: XataAdapter(client),
+  providers: [
+    GoogleProvider({
+      clientId: process.env.GOOGLE_CLIENT_ID,
+      clientSecret: process.env.GOOGLE_CLIENT_SECRET,
+    }),
+  ],
+})
+```
+
+Schema for the Xata Adapter (`@next-auth/xata-adapter`)
+
+## Setup
+
+To get setup with this adapter, you'll need to [create a database](https://docs.xata.io) on Xata and create a schema (a collection of tables with specific structures) that next-auth can use. You can do this using the the Xata CLI. To do this, copy and paste the following schema into a `.json` file somewhere on your file system:
+
+```json title="schema.json"
+{
+  "formatVersion": "",
+  "tables": [
+    {
+      "name": "nextauth_users",
+      "columns": [
+        {
+          "name": "email",
+          "type": "email"
+        },
+        {
+          "name": "emailVerified",
+          "type": "datetime"
+        },
+        {
+          "name": "name",
+          "type": "string"
+        },
+        {
+          "name": "image",
+          "type": "string"
+        },
+        {
+          "name": "username",
+          "type": "string"
+        },
+        {
+          "name": "location",
+          "type": "string"
+        }
```

Comment:
```suggestion
        }
```

To reflect our default models https://next-auth.js.org/adapters/models/

Reply:
Related: https://github.com/nextauthjs/next-auth/pull/4911#discussion_r919812954